quickstart.js

```diff
@@ -66,48 +40,91 @@ async function getAccessToken() {
-  return axios.post(`${baseApiUri}/identityserver/connect/token`, querystring.stringify(payload))
-  .then(function (response) {
-    return response.data;
-  })
-  .catch(function (error) {
-    console.log(error);
-    return null;
+async function getFromApi(serviceAccessToken, resourceUrl, queryParams = {}, raiseError = true) {
+  let api = axios.create({
+    baseURL: baseUrl,
+    headers: {
+      "Authorization": `Bearer ${serviceAccessToken}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json; charset=utf-8"
+    },
+    validateStatus: function (status) {
+      return raiseError ? status === 200 : true;
+    }
   });
+  
+  const url = `${baseUrl}/${resourceUrl}`
+  const response = await api.get(url, { params: queryParams });
+  return response;
+}
+
+// Get a participant access token for the specified participant
+// Used for MyDataHelps Embeddables ONLY
+async function getParticipantAccessToken(serviceAccessToken, participantID, scopes) {
+
+  const payload = {
+      "scope": scopes,
+      "grant_type": "delegated_participant",
+      "participant_id": participantID,
+      "client_id": "MyDataHelps.DelegatedParticipant",
+      "client_secret": "secret",
+      "token": serviceAccessToken,
+  }
+
+  const response = await axios.post(tokenUrl, querystring.stringify(payload));
+  const data = response.data;
+  return data.access_token;
 }
 
+// Use this utility if you ever need to print a response for debugging
 function logResponse(response) {
   console.log(util.inspect(response, { colors: true, depth: 3 }));
 }
 
-getAccessToken().then( token => {
-  if (token) {
-    console.log('Obtained access token:');
-    console.log(token);
-
-    const participantResourceUrl = `/api/v1/administration/projects/` + rksProjectId + '/participants';
-    getFromApi(token, participantResourceUrl)
-    .then(data => {
-      if (data) {
-        // Uncomment this to print out the full response to the console.
-        // logResponse(data);
-        console.log(`\nTotal Participants: ${data.totalParticipants}`);
-      } else {
-        console.log("Error when accessing the API.");
-      }
-    });
-  }
-  else {
-    console.log("Error obtaining access token.");
+async function quickstart() {
+  var url;
+  var response;
+  
+  // Get a service access token, needed for all API calls.
+  const serviceAccessToken = await getServiceAccessToken();
+  console.log('Obtained service access token:');
+  console.log(serviceAccessToken);
+
+  // Get all participants
+  url = `/api/v1/administration/projects/${rksProjectId}/participants`;
+  response = await getFromApi(serviceAccessToken, url);
+  const participants = response.data;
+  console.log(`\nTotal Participants: ${participants.totalParticipants}`);
+
+  // Get a specific participant by identifier. We disable 'raiseError' here
+  // so we can handle the 404 case ourselves.
+  const participantIdentifier = "YOUR_PARTICIPANT_IDENTIFIER"

```

Comment:
What do you think about using participants[0].id?

Reply:
We could, but isn't it better to just prompt them to put in the one they actually want?

Reply:
Sure ... I think my hope was to alleviate confusion between participantID and participantIdentifier so they can figure out which to move... maybe just an in-line comment that says, this will be the id field on the participant object?

Reply:
I'm wondering if we should make it clear that you have to grab the id / specify it here in "YOUR_PARTICIPANT_IDENTIFIER"? Although it's stated pretty clearly in the README (being able to edit the code to specify ppt) I'm just thinking how one could be confused as to what to parse here when one's walking through the steps to use the Embeddables

Reply:
That makes sense. I'll add a comment.